newIDE/app/src/ResourcesList/LocalYarnBridge.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+import optionalRequire from '../Utils/OptionalRequire.js';
+import { type ExternalEditorOpenOptions } from './ResourceExternalEditor.flow';
+import {
+  createOrUpdateResource,
+  getLocalResourceFullPath,
+} from './ResourceUtils.js';
+
+const electron = optionalRequire('electron');
+const path = optionalRequire('path');
+const ipcRenderer = electron ? electron.ipcRenderer : null;
+const gd = global.gd;
+
+/**
+ * Open YARN to Create/Edit Json Dialogue Tree resources.
+ */
+export const openYarn = ({
+  project,
+  resourcesLoader,
+  resourceNames,
+  onChangesSaved,
+  resourcePath,
+  extraOptions,
+}: ExternalEditorOpenOptions) => {
+  if (!electron || !ipcRenderer) return;
+  const projectPath = path.dirname(project.getProjectFile());
+  const initialResourcePath = getLocalResourceFullPath(
+    project,
+    resourceNames[0]
+  );
+
+  const externalEditorData = {
+    resourcePath: initialResourcePath,
+    // $FlowFixMe - TODO: There is an error here to be solved.
```

Comment:
Can you give me the error that is sent here?

Reply:
this is a bit of copy pasta from the one jfxr uses. Not sure if we actually need the flowfixme here tbh, but will test

Reply:
there is actually a flow error if I remove the flowfixme:
```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ src/ResourcesList/LocalYarnBridge.js:46:23

Cannot call onChangesSaved with array literal bound to the first parameter because:
 • property name is missing in object literal [1] but exists in object type [2] in array element.
 • property originalIndex is missing in object literal [1] but exists in object type [2] in array element.

     src/ResourcesList/LocalYarnBridge.js
     43│       const resourceName = path.relative(projectPath, newFilePath);
     44│       createOrUpdateResource(project, new gd.JsonResource(), resourceName);
     45│
 [1] 46│       onChangesSaved([{}], resourceName);
     47│     }
     48│   );
     49│

     src/ResourcesList/ResourceExternalEditor.flow.js
 [2] 14│     Array<{
     15│       path?: string,
     16│       name: string,
     17│       originalIndex: ?number,
     18│       metadata?: ?Object,
     19│     }>,

```

Reply:
See my other comment, if you pass an empty array you should be able to remove this FlowFixMe.
If there is still an error, take a look at it, but there is no reason we need to add FlowFixMe - it's basically a hack and remove the value of having Flow checking the types.